usr/share/rear/backup/NETFS/default/500_make_backup.sh

```diff
@@ -71,45 +71,91 @@ LogPrint "Creating $BACKUP_PROG archive '$backuparchive'"
-			$BACKUP_PROG_CREATE_NEWER_OPTIONS \
-			${BACKUP_PROG_BLOCKS:+-b $BACKUP_PROG_BLOCKS} "${BACKUP_PROG_COMPRESS_OPTIONS[@]}" \
-			-X $TMP_DIR/backup-exclude.txt -C / -c -f - \
-			$(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE \| $BACKUP_PROG_CRYPT_OPTIONS BACKUP_PROG_CRYPT_KEY \| $SPLIT_COMMAND
-		$BACKUP_PROG $TAR_OPTIONS --sparse --block-number --totals --verbose \
-			--no-wildcards-match-slash --one-file-system \
-			--ignore-failed-read "${BACKUP_PROG_OPTIONS[@]}" \
-			$BACKUP_PROG_CREATE_NEWER_OPTIONS \
-			${BACKUP_PROG_BLOCKS:+-b $BACKUP_PROG_BLOCKS} "${BACKUP_PROG_COMPRESS_OPTIONS[@]}" \
-			-X $TMP_DIR/backup-exclude.txt -C / -c -f - \
-			$(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE | $BACKUP_PROG_CRYPT_OPTIONS $BACKUP_PROG_CRYPT_KEY | $SPLIT_COMMAND
-	;;
-	(rsync)
-		# make sure that the target is a directory
-		mkdir -p $v "$backuparchive" >&2
-		Log $BACKUP_PROG --verbose "${BACKUP_RSYNC_OPTIONS[@]}" --one-file-system --delete \
-			--exclude-from=$TMP_DIR/backup-exclude.txt --delete-excluded \
-			$(cat $TMP_DIR/backup-include.txt) "$backuparchive"
-		$BACKUP_PROG --verbose "${BACKUP_RSYNC_OPTIONS[@]}" --one-file-system --delete \
-			--exclude-from=$TMP_DIR/backup-exclude.txt --delete-excluded \
-			$(cat $TMP_DIR/backup-include.txt) "$backuparchive" >&2
-	;;
-	(*)
-		Log "Using unsupported backup program '$BACKUP_PROG'"
-		Log $BACKUP_PROG "${BACKUP_PROG_COMPRESS_OPTIONS[@]}" \
-			$BACKUP_PROG_OPTIONS_CREATE_ARCHIVE $TMP_DIR/backup-exclude.txt \
-			"${BACKUP_PROG_OPTIONS[@]}" $backuparchive \
-			$(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE > $backuparchive
-		$BACKUP_PROG "${BACKUP_PROG_COMPRESS_OPTIONS[@]}" \
-			$BACKUP_PROG_OPTIONS_CREATE_ARCHIVE $TMP_DIR/backup-exclude.txt \
-			"${BACKUP_PROG_OPTIONS[@]}" $backuparchive \
-			$(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE > $backuparchive
-	;;
+    # tar compatible programs here
+    (tar)
+        set_tar_features
+        Log $BACKUP_PROG $TAR_OPTIONS --sparse --block-number --totals --verbose \
+            --no-wildcards-match-slash --one-file-system \
+            --ignore-failed-read "${BACKUP_PROG_OPTIONS[@]}" \
+            $BACKUP_PROG_CREATE_NEWER_OPTIONS \
+            ${BACKUP_PROG_BLOCKS:+-b $BACKUP_PROG_BLOCKS} "${BACKUP_PROG_COMPRESS_OPTIONS[@]}" \
+            -X $TMP_DIR/backup-exclude.txt -C / -c -f - \
+            $(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE \| $BACKUP_PROG_CRYPT_OPTIONS BACKUP_PROG_CRYPT_KEY \| $SPLIT_COMMAND
+
+        # Variable used to record the short name of piped commands in case of
+        # error, e.g. ( "tar" "cat" "dd" ) in case of unencrypted and unsplit backup.
+        backup_prog_shortnames=(
+            "$(basename $(echo "$BACKUP_PROG" | awk '{ print $1 }'))"
+            "$(basename $(echo "$BACKUP_PROG_CRYPT_OPTIONS" | awk '{ print $1 }'))"
+            "$(basename $(echo "$SPLIT_COMMAND" | awk '{ print $1 }'))"
+        )
+        for index in ${!backup_prog_shortnames[@]} ; do
```

Comment:
Using `${!array_variable[@]}` may not work on older systems.

On my SLES11 with bash 3.2 this is not mentioned in `man bash`
but it works nevertheless.

I don't know if bash on older RHEL support it.

If it is not supported you may do it as follows:
<pre>
prog_basename="$(basename $(echo "$BACKUP_PROG" | awk '{ print $1 }'))"
test $prog_basename || BugError "Failed to get basename for pipe component '$BACKUP_PROG'"

crypt_basename="$(basename $(echo "$BACKUP_PROG_CRYPT_OPTIONS" | awk '{ print $1 }'))"
test $crypt_basename || BugError "Failed to get basename for pipe component '$BACKUP_PROG_CRYPT_OPTIONS'"

split_basename="$(basename $(echo "$SPLIT_COMMAND" | awk '{ print $1 }'))"
test $split_basename || BugError "Failed to get basename for pipe component '$SPLIT_COMMAND'"

backup_prog_shortnames=( "$prog_basename" "$crypt_basename" "$split_basename" )
</pre>

Reply:
rhel6 ships bash4.1 so it works. that `!array[@]` syntax is quite old already.
This was introduced in bash3.0

Reply:
But RHEL 5 is also still supported, see
http://relax-and-recover.org/documentation/release-notes-2-4
and for the upcoming ReaR 2.5 see
https://github.com/rear/rear.github.com/blob/master/documentation/release-notes-2-5.md

Reply:
works also with rhel5 (bash 3.2)